### ISSUE TITLE
termux-pacman

### ISSUE THREAD

**Issue by Freed-Wu** [ustclug/mirrorrequest#348]
### 项目介绍 / Project Introduction

Support for pacman packages in termux

### 上游地址及镜像方式 / Upstream location & mirror method

https://termux-pacman.github.io/

### 为什么希望添加该镜像 / Why do you want we to add this mirror

ustclug has owned a mirror of termux. May be termux-pacman can also be supported? TIA!

### 其他信息 / Extra information

- 镜像容量（Mirror size）：
- 该镜像需求是否曾在之前的 issue 中提出过：No

Related issues:

- tur:
  - https://github.com/ustclug/mirrorrequest/issues/360
  - https://github.com/tuna/issues/issues/1906
- termux-pacman
  - https://github.com/tuna/issues/issues/1635

**Comment by iBug** [comment]
Will you elaborate on the differences between termux-pacman and the official APT repository? It seems to me, at first glance, mostly duplicate content (only packaged in another format).

**Comment by Maxython** [comment]
> Will you elaborate on the differences between termux-pacman and the official APT repository? It seems to me, at first glance, mostly duplicate content (only packaged in another format).

Even though the pacman service and apt have the same content, it is a bit wrong to think that the pacman service is a duplicate (there are 3 main reasons).

1. The pacman service has termux packages in the pacman format (which is obvious and logical), but the addition of these packages is done from scratch, that is, these packages are completely compiled in the forked [termux-packages](https://github.com/termux-pacman/termux-packages) repository.
2. The pacman service has its own signature for packages and for files that work with pacman (these are `db` and `files`). [Signature link](https://keyserver.ubuntu.com/pks/lookup?search=termux-pacman&fingerprint=on&op=index).
3. In the pacman service, in addition to the usual repositories from Termux (main, x11 and root), there are also repositories from [TUR](https://github.com/termux-user-repository) (Termux User Repository). These are [tur](https://s3.amazonaws.com/termux-pacman.us/tur/index.html) and [tur-continuous](https://s3.amazonaws.com/termux-pacman.us/tur-continuous/index.html).

Hope this answers your question. And since I manage pacman services, I wanted to know more about your mirror service.

**Comment by iBug** [comment]
Hi @Maxython, thank you for the explanation. My considerations are:

1. **How many users this new mirror repository can benefit.** Since our storage capacity isn't infinite, we want our service to benefit as many users as possible. So in general, we favor projects with a larger user base than those highly domain-specific.

   At present, I have no idea how popular termux-pacman is, compared to other projects that already have a mirror request pending.

2. **Whether it is *substantially* different from an existing repository.** We've decided that most RHEL clones (CentOS, AlmaLinux, Rocky Linux etc.) are not, and put all requests about them on hold.

   I am not convinced that the usual repositories (main, x11, root) are substantially different from the official deb packages. Until this can be demonstrated, it's unlikely that we look favorably upon this project as a mirroring candidate.

**Comment by Maxython** [comment]
> At present, I have no idea how popular termux-pacman is, compared to other projects that already have a mirror request pending.

It's hard for me to say how popular the pacman service (or more precisely the pacman-for-termux project) is, since I don't do any analysis. But I can say with confidence that the project is becoming more and more popular.

> I am not convinced that the usual repositories (main, x11, root) are substantially different from the official deb packages. Until this can be demonstrated, it's unlikely that we look favorably upon this project as a mirroring candidate.

The [main](https://s3.amazonaws.com/termux-pacman.us/main/index.html), [x11](https://s3.amazonaws.com/termux-pacman.us/x11/index.html) and [root](https://s3.amazonaws.com/termux-pacman.us/root/index.html) repositories contain the same packages as in the official termux service, only everything is in pacman format and signed by termux-pacman.

To be honest, the pacman service isn't ready to be mirrored yet. It needs to be finalized to a full-fledged site.

**Comment by Maxython** [comment]
And so, here are some statistics for you so that you understand how popular the project is:
-  Chine (13.01-25.01):
   - Number of requests: 1067, this is 6.42% of the number of requests worldwide (5th place out of 50)
   - Number by download MB: 238.95 (7th out of 50)
 - Asia (if your mirrors are also for neighboring countries, 13.01-25.01):
   - Number of requests: 5486, this is approximately 33.17% of the world's queries.
   - Number by download GB: 2.47, this is approximately 44.58% of the world's number of downloads.

**Comment by Maxython** [comment]
Are there any news?

**Comment by iBug** [comment]
It seems like this repository has very scarce interest and utility. At this level of activity, it's unlikely that we will consider mirroring this sometime soon. I'll leave this open for future reference.

**Comment by iBug** [comment]
Closing for insufficient general interest and activity.

**Comment by Maxython** [comment]
Just for my understanding: did you add the label `hard to sync` because it is hard (slow) to connect to the rsync server [`sync.termux-pacman.dev`](https://termux-pacman.dev/mirroring/) from China?

**Comment by taoky** [comment]
> Just for my understanding: did you add the label `hard to sync` because it is hard (slow) to connect to the rsync server [`sync.termux-pacman.dev`](https://termux-pacman.dev/mirroring/) from China?

No, just because we have no idea that a rsync upstream exists (and syncing through HTTP(S) is not trivial).
